Content.Server/GameObjects/Components/Mobs/Body/BodyTemplates/BodyTemplates.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using System.Collections.Generic;
+using Content.Shared.GameObjects;
+using Content.Server.GameObjects.Components.Mobs.Body.Organs;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.GameObjects.Components.Mobs.Body
+{
+    /// <summary>
+    ///    Core of the mobcode. It glues all the shitcode with limbs, organs 
+    ///    and body functions together with DAMAGE, making frankensteins that we call Mobs
+    /// </summary>
+
+    public class BodyTemplate
+    {
+#pragma warning disable CS0649
+        [Dependency]
+        private IPrototypeManager PrototypeManager;
+#pragma warning restore
+
+        public string Name { get; private set; }
+        public string Id { get; private set; }
+        public float TimeSinceUpdate { get; private set; }
+
+        /// <summary>
+        /// All body parts - limbs and organs alike. From this list Life() is called.
+        /// </summary>
+        protected List<BodyPart> AllBodyParts;
+        /// <summary>
+        /// The list of all mob's limbs. From this list damage is spreading. 
+        /// </summary>
+        protected List<Limb> BodyMap;
+        /// <summary>
+        /// helper list to assign children to BodyMap
+        /// </summary>
+        private List<Limb> Limbs;
+
+
+
+        /// <summary>
+        /// Entity which owns MobComponent
+        /// </summary>
+        public IEntity Owner;
+
+        /// <summary>
+        /// List of YAML strings which inits all the limbs
+        /// </summary>
+        private List<string> _limbProts;
+
+        Random _randomLimb
+        {
+            get 
+            {
+                return new Random(Owner.Uid.GetHashCode() ^ DateTime.Now.GetHashCode());
```

Comment:
it's MORE random

Reply:
No, it's not.

If I do this:

```cs
var a = _seed.Next();
var b = _seed.Next();
```

There is a very good chance for A and B to be the exact same value, because the PRNG gets the same seed. Don't do this.

Reply:
what about this
```csharp
int counter = 0;
Random _seed
{
get 
     {
            counter++;
            return new Random(Owner.Uid.GetHashCode() ^ DateTime.Now.GetHashCode()^counter);
     }
}
```